src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,681 @@
+    var segs = ["contexts", context, "preferences", product.replace(/\./g, "\\.")];
+    var path = "flatPrefs";
+    fluid.each(segs, function (item) { path += "." + item; });
+
+    if (checked) {
+        // Add a check in case it's already enabled
+        that.applier.change(path, {});
+        that.addEditToUnsavedList("Enabled Product for Context: " + product);
+    }
+    else {
+        that.applier.change(path, false, "DELETE");
+        that.addEditToUnsavedList("Un-enabled Product for Context: " + product);
+    }
+    that.applier.change("npsetApplications", gpii.devpmt.npsetApplications(that.model.flatPrefs));
+    that.applier.change("unsavedChangesExist", true);
+};
+
+/**
+ * Event Listener for when a button is clicked to delete
+ * a context. Fetches the context to delete from the elements
+ * data-contextid attribute.
+ *
+ * This listener doesn't actually delete the context, but
+ * opens the confirm delete context dialog.
+ *
+ * @param event (Event) Browser event object
+ */
+gpii.devpmt.onDeleteContext = function (that, event) {
+    var contextId = event.currentTarget.dataset.contextid;
+    that.applier.change("activeModalDialog.contextId", contextId);
+    that.events.openConfirmDeleteContextDialog.fire();
+};
+
+gpii.devpmt.editValueEvent = function (that, event) {
+    var newCurrent = {
+        context: event.currentTarget.dataset.context,
+        term: event.currentTarget.dataset.term,
+        //value: event.currentTarget.dataset.value,
+        product: event.currentTarget.dataset.product
+    };
+
+    var newMetadata = {};
+    if (!newCurrent.product) {
+        // Common Term
+        newMetadata = that.options.commonTerms[newCurrent.term];
+        newMetadata.name = newMetadata.schema.title;
+        newMetadata.description = newMetadata.schema.description;
+        newCurrent.value = that.lookupGenericPrefValue(newCurrent.context, newCurrent.term);
+        newCurrent.blank = newCurrent.value === undefined;
+    }
+    else {
+        // Application Specific
+        console.log(newCurrent);
+        // TODO ontology
+        newMetadata = gpii.devpmt.findProductSettingMetadata(that.model.allSolutions,
+            newCurrent.product.slice(38), newCurrent.term);
+        newMetadata.name = newMetadata.schema.title;
+        newMetadata.description = newMetadata.schema.description;
+        newCurrent.value = that.lookupProductPrefValue(newCurrent.context, newCurrent.product, newCurrent.term);
+        newCurrent.blank = newCurrent.value === undefined;
+    }
+    that.applier.change("currentlyEditing.current", newCurrent);
+    that.applier.change("currentlyEditing.metadata", newMetadata);
+    that.applier.change("currentlyEditing.active", true);
+    that.applier.change("unsavedChangesExist", true);
+    that.prefsAdjuster.renderInitialMarkup();
+    // $(document).foundation();
+    // https://github.com/zurb/foundation-sites/issues/7899
+    $(".sticky").foundation("_calc", true);
```

Comment:
Won't this get done automatically by your markup updated listener?

Reply:
this instance has been removed